Formula/fastd.rb

```diff
@@ -26,21 +26,35 @@ class Fastd < Formula
   end
 
   depends_on "cmake" => :build
-  depends_on "libuecc"
-  depends_on "libsodium"
   depends_on "bison" => :build # fastd requires bison >= 2.5
   depends_on "pkg-config" => :build
   depends_on "json-c"
+  depends_on "libuecc"
   depends_on "openssl" => :optional
   depends_on :tuntap => :optional
 
+  # https://projects.universe-factory.net/issues/251
+  resource "libsodium" do
+    url "https://github.com/jedisct1/libsodium/releases/download/1.0.14/libsodium-1.0.14.tar.gz"
+    sha256 "3cfc84d097fdc891b40d291f2ac2c3f99f71a87e36b20cc755c6fa0e97a77ee7"
+  end
+
   def install
     args = std_cmake_args
     args << "-DENABLE_LTO=ON"
     args << "-DENABLE_OPENSSL=ON" if build.with? "openssl"
-    args << buildpath
+
+    resource("libsodium").stage do
```

Comment:
It seems weird it only broke one of them. Have you tried to patch it or drop the dependency?

Reply:
`libsodium` upstream noted basically they couldn't find anyone using the `aes128ctr` code & it had various undesirable attributes so it got culled. I guess they hadn't discovered `fastd`, heh.

You can turn the dependency off with a flag, if preferred over a temporary vendor job.

Reply:
```
Josephs-MacBook-Pro:~ joe$ brew formula-analytics fastd
install events in the last 30 days for fastd
=====================================================================================================
1 | fastd                                                                              | 18 |  94.74%
2 | fastd --with-openssl                                                               |  1 |   5.26%
=====================================================================================================
Total                                                                                  | 19 |    100%
=====================================================================================================

```
I'd just drop the dep. I doubt we'll hear a peep.

Reply:
Will do 👍.

Reply:
Lol great, there's a mandatory fallback, apparently.

```
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
NACL_INCLUDE_DIR
   used as include directory in directory /tmp/fastd-20171002-27136-1ilmkta/fastd-18/src
   used as include directory in directory /tmp/fastd-20171002-27136-1ilmkta/fastd-18/src/methods/generic_poly1305
   used as include directory in directory /tmp/fastd-20171002-27136-1ilmkta/fastd-18/src/methods/xsalsa20_poly1305
   used as include directory in directory /tmp/fastd-20171002-27136-1ilmkta/fastd-18/src/crypto/cipher/aes128_ctr/nacl
   used as include directory in directory /tmp/fastd-20171002-27136-1ilmkta/fastd-18/src/crypto/cipher/salsa2012/nacl
   used as include directory in directory /tmp/fastd-20171002-27136-1ilmkta/fastd-18/src/crypto/cipher/salsa20/nacl
NACL_LIBRARY
    linked by target "fastd" in directory /tmp/fastd-20171002-27136-1ilmkta/fastd-18/src
    linked by target "method_generic_poly1305" in directory /tmp/fastd-20171002-27136-1ilmkta/fastd-18/src/methods/generic_poly1305
    linked by target "method_xsalsa20_poly1305" in directory /tmp/fastd-20171002-27136-1ilmkta/fastd-18/src/methods/xsalsa20_poly1305
    linked by target "cipher_aes128_ctr_nacl" in directory /tmp/fastd-20171002-27136-1ilmkta/fastd-18/src/crypto/cipher/aes128_ctr/nacl
    linked by target "cipher_salsa2012_nacl" in directory /tmp/fastd-20171002-27136-1ilmkta/fastd-18/src/crypto/cipher/salsa2012/nacl
    linked by target "cipher_salsa20_nacl" in directory /tmp/fastd-20171002-27136-1ilmkta/fastd-18/src/crypto/cipher/salsa20/nacl
```

Reply:
Tried lying to it but the build blew up:
```
      -DENABLE_LIBSODIUM=OFF
      -DNACL_INCLUDE_DIR=""
      -DNACL_LIBRARY=""
```

```
[  8%] Building C object src/crypto/cipher/null/CMakeFiles/cipher_null.dir/null.c.o
cd /tmp/fastd-20171002-32993-gh2vmc/fastd-18/fastd-build/src/crypto/cipher/null && /usr/local/Homebrew/Library/Homebrew/shims/super/clang -D_GNU_SOURCE -D__APPLE_USE_RFC_3542 -I/tmp/fastd-20171002-32993-gh2vmc/fastd-18 -I/tmp/fastd-20171002-32993-gh2vmc/fastd-18/fastd-build/gen  -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk -mmacosx-version-min=10.13   -pthread -std=c99    -flto -Wall -o CMakeFiles/cipher_null.dir/null.c.o   -c /tmp/fastd-20171002-32993-gh2vmc/fastd-18/src/crypto/cipher/null/null.c
/tmp/fastd-20171002-32993-gh2vmc/fastd-18/src/crypto/cipher/salsa20/nacl/salsa20_nacl.c:36:10: fatal error: 'crypto_stream_salsa20.h' file not found
#include <crypto_stream_salsa20.h>
         ^~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/fastd-20171002-32993-gh2vmc/fastd-18/src/crypto/cipher/aes128_ctr/nacl/cipher_aes128_ctr_nacl.c:36:10: fatal error: 'crypto_stream_aes128ctr.h' file not found
#include <crypto_stream_aes128ctr.h>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~
11 error generated.
 error generated.
make[2]: *** [src/crypto/cipher/aes128_ctr/nacl/CMakeFiles/cipher_aes128_ctr_nacl.dir/cipher_aes128_ctr_nacl.c.o] Error 1
make[2]: *** [src/crypto/cipher/salsa20/nacl/CMakeFiles/cipher_salsa20_nacl.dir/salsa20_nacl.c.o] Error 1
make[1]: *** [src/crypto/cipher/aes128_ctr/nacl/CMakeFiles/cipher_aes128_ctr_nacl.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
make[1]: *** [src/crypto/cipher/salsa20/nacl/CMakeFiles/cipher_salsa20_nacl.dir/all] Error 2
```

Reply:
OK let's see if upstream responds. I'd rather remove the formula than vendor it.

Reply:
Just skip the revision bump on this one. It's ok to break it.

Reply:
Bump dropped.

Reply:
As in, remove all of the changes to that file.

Reply:
Including the vendor/bug report/etc?

Reply:
yup.

Reply:
Alright. For posterity, https://projects.universe-factory.net/issues/251 is the upstream bug report.